sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -164,6 +156,20 @@ public boolean start() throws IOException {
     // since CPU consumed on the workers would be low and will likely avoid unnecessary upscale.
     offsetConsumerConfig.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, "read_uncommitted");
 
+    if (spec.getOffsetConsumerConfig() == null) {
```

Comment:
What if config does not set groupId(). I think we should keep the earlier code to assign randomized groupid and let it be overwritten by the user.

Reply:
make sense to avoid impact on the main consumer.